nexus/db-model/src/semver_version.rs

```diff
@@ -0,0 +1,64 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use diesel::backend::{Backend, RawValue};
+use diesel::deserialize::{self, FromSql};
+use diesel::query_builder::bind_collector::RawBytesBindCollector;
+use diesel::serialize::{self, ToSql};
+use diesel::sql_types;
+use omicron_common::api::external;
+use parse_display::Display;
+use serde::{Deserialize, Serialize};
+
+// We wrap semver::Version in external to impl JsonSchema, and we wrap it again
+// here to impl ToSql/FromSql
+
+#[derive(
+    Clone,
+    Debug,
+    AsExpression,
+    FromSqlRow,
+    Serialize,
+    Deserialize,
+    PartialEq,
+    Display,
+)]
+#[diesel(sql_type = sql_types::Text)]
+#[display("{0}")]
+pub struct SemverVersion(pub external::SemverVersion);
+
+NewtypeFrom! { () pub struct SemverVersion(external::SemverVersion); }
+NewtypeDeref! { () pub struct SemverVersion(external::SemverVersion); }
+
+impl SemverVersion {
+    pub fn new(major: u64, minor: u64, patch: u64) -> Self {
```

Comment:
Will we ever make it possible to have either "prerelease" or "build metadata" as a string here? 
I'm eyeing the https://docs.rs/semver/latest/semver/struct.Version.html struct for comparison.

(IMO if we say "no for this PR", I'm totally fine with that, just wondering about future expansion)

Reply:
Yeah, I think so. You could always use the struct constructor directly, though in our case that's pretty annoying, as we have two different wrappers. So we'd probably just add more constructor functions.